luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -0,0 +1,224 @@
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+ */
+
+package org.opencadc.luskan.ws;
+
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.db.DBUtil;
+import org.opencadc.luskan.impl.DataSourceProviderImpl;
+import org.opencadc.luskan.impl.InitLuskanSchemaContent;
+import ca.nrc.cadc.reg.client.LocalAuthority;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.rest.RestAction;
+import ca.nrc.cadc.tap.schema.InitDatabaseTS;
+import ca.nrc.cadc.uws.server.impl.InitDatabaseUWS;
+import ca.nrc.cadc.vosi.AvailabilityPlugin;
+import ca.nrc.cadc.vosi.AvailabilityStatus;
+import ca.nrc.cadc.vosi.avail.CheckDataSource;
+import ca.nrc.cadc.vosi.avail.CheckException;
+import ca.nrc.cadc.vosi.avail.CheckResource;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.Principal;
+import javax.security.auth.Subject;
+import javax.security.auth.x500.X500Principal;
+import javax.sql.DataSource;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author adriand
+ */
+public class StorageTapService implements AvailabilityPlugin {
+
+    private static final Logger log = Logger.getLogger(StorageTapService.class);
+    
+    private static final Principal TRUSTED = new X500Principal("cn=servops_4a2,ou=cadc,o=hia,c=ca");
+
+    private static final String TAP_TEST = "select schema_name from tap_schema.schemas11 where schema_name='tap_schema'";
+    private static final String UWS_TEST = "select jobID from uws.Job limit 1";
+
+    private String appName;
+
+    public StorageTapService() {
+    }
+
+    @Override
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+    
+    @Override
+    public AvailabilityStatus getStatus() {
+        boolean isGood = true;
+        String note = "service is accepting queries";
+        try {
+            String state = getState();
+            if (RestAction.STATE_OFFLINE.equals(state)) {
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_OFFLINE_MSG);
+            }
+            if (RestAction.STATE_READ_ONLY.equals(state)) {
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_READ_ONLY_MSG);
+            }
+
+            String tapDsName = DataSourceProviderImpl.getDataSourceName(null, "tapadm");
```

Comment:
Since we are using the default data source names:

```
DataSource tapadm = DBUtil.findJNDIDataSource("jdbc/tapadm");
DataSource uws = DBUtil.findJNDIDataSource("jdbc/uws");
DataSource tapuser = DBUtil.findJNDIDataSource("jdbc/tapuser");
```

The CheckDataSource(uwsDsName, UWS_TEST) is redundant when you used InitDatabaseUWS.

The "run a couple of queries" below MUST use the tapuser pool. In that case, you need at least one test query in tap_schema and one in inventory (to be sure tapuser has permission to select from them). Checking all tables is not useful since the grants are generally (the libs do "grant select on schema .... to public").

Reply:
All done except "select from inventory" since content of the inventory is not created by luskan